src/NServiceBus.Core/Unicast/ContextualBus.cs

```diff
@@ -270,11 +271,21 @@ public void HandleCurrentMessageLater()
                 return;
             }
 
-            messageSender.Send(new OutgoingMessage(MessageBeingProcessed.Id, MessageBeingProcessed.Headers, MessageBeingProcessed.Body), new TransportSendOptions(sendLocalAddress));
+            var context = incomingContext as HandlingStageBehavior.Context;
+            if (context != null)
```

Comment:
Nitpick: I personally prefer to exit early:

```
if(context == null)
{
  log.Info("Noop for HandleCurrentMessageLater because there is no incoming message.");
  return;
}

```

Think the code becomes more clear?